src/main/java/org/broadinstitute/hellbender/tools/GetSampleName.java

```diff
@@ -0,0 +1,65 @@
+package org.broadinstitute.hellbender.tools;
+
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.BetaFeature;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.QCProgramGroup;
+import org.broadinstitute.hellbender.engine.GATKTool;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@CommandLineProgramProperties(
+        oneLineSummary = "(EXPERIMENTAL) Emit a single sample name from the bam header into an output file.",
+        summary = "If the bam has zero or more than one sample names in the header, this tool will error, by design.\n" +
+                "  This tool has not been tested extensively.  Most options supported by the GATK are irrelevant for this tool.",
+        programGroup = QCProgramGroup.class
+)
+@BetaFeature
+final public class GetSampleName extends GATKTool{
+
+    @Argument(
+            doc = "Output file with only the sample name in it.",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME
+    )
+    protected File outputSampleNameFile;
+
+    @Override
+    public void traverse() {
+        // Do nothing!
+    }
+
+    @Override
+    public boolean requiresReads() {return true;}
+
+    @Override
+    public void onTraversalStart() {
+
+        // Grab the header info
+        if ((getHeaderForReads() == null) || (getHeaderForReads().getReadGroups() == null)) {
+            throw new UserException.BadInput("The given input bam has no header or no read groups.  Cannot determine a sample name.");
+        }
+
+        final List<String> sampleNames = getHeaderForReads().getReadGroups().stream().map(s -> s.getSample()).distinct().collect(Collectors.toList());
```

Comment:
This could be simplified slightly if you collect as `Set` instead of `List`, which will let you avoid `distinct()`. Also I would prefer `SAMReadGroupRecord::getSample` over `s -> s.getSample()` for readability even though the lambda form is fewer characters.